File: server/models/Workflow.js
const mongoose = require('mongoose');

const SubStepSchema = new mongoose.Schema({
    type: String,
    value: String // The text content of the sub-step
});

const StepSchema = new mongoose.Schema({
    stepId: Number,
    title: String,
    description: String,
    subSteps: [String], // Array of strings based on current structure
    documents: [String],
    source: String
});

const WorkflowSchema = new mongoose.Schema({
    goal: { type: String, required: true },
    steps: [StepSchema],
    locationContext: {
        origin: String,
        destination: String,
        query: String
    },
    confidenceScore: Number,
    guestId: { type: String, index: true }, // Added for guest isolation
    language: String, // Added to track language
    createdAt: { type: Date, default: Date.now }
});

module.exports = mongoose.model('Workflow', WorkflowSchema);

File: server/models/workflow.py
from pydantic import BaseModel, Field
from datetime import datetime
from typing import List, Optional

class LocationContext(BaseModel):
    origin: Optional[str] = None
    destination: Optional[str] = None
    query: Optional[str] = None

class StepModel(BaseModel):
    stepId: int
    title: str
    description: str
    subSteps: List[str] = []
    documents: List[str] = []
    source: Optional[str] = None

class WorkflowModel(BaseModel):
    goal: str
    steps: List[StepModel] = []
    locationContext: Optional[LocationContext] = None
    confidenceScore: Optional[int] = None
    guestId: str
    language: Optional[str] = 'English'
    createdAt: datetime = Field(default_factory=datetime.utcnow)
